### PR TITLE
Print a better error message when the expected backup site is missing

### PR DESCRIPTION
**Pull request by jelmer**
This seems slightly better than a plain KeyError. Is there something better we can do in this situation?

### CODE REVIEW

File: myhoard/controller.py [hunk 1]
```diff
@@ -669,7 +669,12 @@ def _binlog_uploaded(self, *, local_index, remote_key, stream):
 
     def _build_backup_stream(self, backup):
         stream_id = backup["stream_id"]
-        backup_site = self.backup_sites[backup["site"]]
+        backup_site_name = backup["site"]
+        try:
+            backup_site = self.backup_sites[backup_site_name]
+        except KeyError:
+            self.stats.increase("myhoard.unknown_backup_site", tags={"backup_site": backup_site_name})
+            raise AssertionError(f"backup site {backup_site_name} unknown (sites: {list(self.backup_sites.keys())!r})")
```
Comment (packi) [inline myhoard/controller.py:677]: I would change this to a specific exception so if this ends up in a log it's clear from the exception what could happen. Helps with classifying errors without stats.

File: myhoard/controller.py [hunk 2]
```diff
@@ -774,7 +779,12 @@ def _create_restore_coordinator_if_missing(self):
             return
 
         options = self.state["restore_options"]
-        backup_site = self.backup_sites[options["site"]]
+        backup_site_name = options["site"]
+        try:
+            backup_site = self.backup_sites[backup_site_name]
+        except KeyError:
+            self.stats.increase("myhoard.unknown_backup_site", tags={"backup_site": backup_site_name})
+            raise AssertionError(f"backup site {backup_site_name} unknown (sites: {list(self.backup_sites.keys())!r})")
```
Comment (packi) [inline myhoard/controller.py:787]: Same here.

**Review by packi — DISMISSED** [review]
NIT: I would probably extract the lookup into a function so we're sure that the exception and stats match in the error case.

File: myhoard/controller.py (hunk 1 above)
Comment (jelmer) [reply myhoard/controller.py:677]: Done.

File: myhoard/controller.py (hunk 2 above)
Comment (jelmer) [reply myhoard/controller.py:787]: Done.

**Review by packi — APPROVED** [review]

**Pull request merged**